mx.c

```diff
@@ -255,8 +255,22 @@ struct Context *mx_mbox_open(struct Mailbox *m, OpenMailboxFlags flags)
   if (!m)
     return NULL;
 
-  struct Context *ctx = ctx_new();
-  ctx->mailbox = m;
+  struct Context *ctx = NULL;
+  struct Context *np = NULL;
+  STAILQ_FOREACH(np, &contexts, entries)
+  {
+    if (np->mailbox == m)
```

Comment:
Is pointer equality sufficient here? Do we assume that if we already have a mailbox, we'll always find it by the same pointer? Maybe that is correct. Should we also return an existing mailbox if their path match? Maybe, too..

Reply:
Yes, I think comparing pointers should be enough here, we can always extend the usage if necessary. In case you're curious on how I intend to use this have a look at
https://github.com/sassmann/neomutt/commit/d35aa695098b2dd941fda783c000da89425fe608